linkerd/app/core/src/serve.rs

```diff
@@ -52,7 +52,13 @@ where
                                         .await
                                     {
                                         Ok(()) => debug!("Connection closed"),
-                                        Err(reason) => debug!(%reason, "Connection closed"),
+                                        Err(reason) => {
+                                            if is_io(&*reason) {
+                                                debug!(%reason, "Connection closed")
+                                            } else {
+                                                info!(%reason, "Connection closed")
+                                            }
```

Comment:
TIOLI, might be a little bit more concise:
```suggestion
                                        Err(reason) if is_io(&*reason) => debug!(%reason, "Connection closed"),
                                        Err(reason) => info!(%reason, "Connection closed"),
```